libretro/libretro.cpp

```diff
@@ -705,19 +714,32 @@ struct SaveStart {
 } // namespace SaveState
 
 size_t retro_serialize_size(void) {
+	// Not pausing the EmuThread here should be somewhat safe, for the time being
```

Comment:
If retro_serialize_size and retro_serialize are called together to find out the size of memory and then serialize, you should try to get them inside the same "EmuThreadPause/EmuThreadStart" block...

Reply:
I did consider just pausing it there and then remove the pause from serialize (or just leaving it since it already checks if its paused), but I wouldn't be able to continue there, because it would advance the frames.
It's the same issue in the other serialize functions, EmuThreadPause will advance by a frame, so it's already guaranteed to be out of sync.
The real question is if the Savestate Buffer ever shrinks, since only that should cause actual Issues?
Alternative is a hard crash right now, either way.
But I can look into that after work if you want.

Reply:
Now that I think about it, I should probably wrap them in if(useEmuThread) too, since it's only enabled for GL it seems.
It should be required for all renderer`'s anyway, right?